CodaClient.py

```diff
@@ -6,6 +6,89 @@
 import asyncio
 import websockets
 import logging
+from enum import Enum
+
+class CurrencyFormat(Enum):
+  WHOLE = 1
+  NANO = 2
+
+class CurrencyUnderflow(Exception):
+  pass
+
+class Currency():
+  @classmethod
+  def __nanocodas_from_int(_cls, n):
+    return n * 1000000000
+
+  @classmethod
+  def __nanocodas_from_string(_cls, s):
+    segments = s.split('.')
+    if len(segments) == 1:
+      return int(segments[0])
+    elif len(segments) == 2:
+      [l, r] = segments
+      if len(r) <= 9:
+        return int(l + r + ('0' * (9 - len(r))))
+      else:
+        raise Exception('invalid coda currency format: %s' % s)
+
+  def __init__(self, value, format=CurrencyFormat.WHOLE):
+    if format == CurrencyFormat.WHOLE:
+      if isinstance(value, int):
+        self.__nanocodas = Currency.__nanocodas_from_int(value)
+      elif isinstance(value, float):
+        self.__nanocodas = Currency.__nanocodas_from_string(str(value))
+      elif isinstance(value, str):
+        self.__nanocodas = Currency.__nanocodas_from_string(value)
+      else:
+        raise Exception('cannot construct whole Currency from %s' % type(value))
+    elif format == CurrencyFormat.NANO:
+      if isinstance(value, int):
+        self.__nanocodas = value
+      else:
+        raise Exception('cannot construct nano Currency from %s' % type(value))
+    else:
+      raise Exception('invalid Currency format %s' % format)
+
+  def decimal_format(self):
+    s = str(self.__nanocodas)
+    if len(s) > 9:
+      return s[:-9] + '.' + s[-9:]
```

Comment:
This will have trailing zeroes which won't match the formatting done by the daemon/wallet, but it's probably enough for debugging purposes